### ISSUE TITLE
WPF Apps Crash when clicking main menu options (crash in DXD9.dll)

### ISSUE THREAD

**Issue by Louisdegel** [Ryder25/Ego-Engine-Modding#27]
Programm crashes when i click File, Recources etc.

**Comment by Filipe-Misael** [comment]
I have the exact same problem.
I've tried all the suggestions on the page, including starting in admin mode, compatibility mode (tried OS option)

A few more details that might help figure out the problem:

The software worked just fine on this same computer, it stopped working after I recently formatted (And windows is fully updated) I had windows 10 ultimate before, I'm now using windows 10 workstation (Not sure if that's the problem!)

I can open pssg files with your software (properties > open with etc...) and I can view the images but as soon as I try to click on any of the buttons above, like "File" the software just freezes and then closes itself about 2 seconds later... So I can't import or export anything.
There's no error message...

(And yes, I did try re-downloading the mod tool and even save it on a different hard drive, it's the exact same problem)


> 
> 
> Programm crashes when i click File, Recources etc.

@Louisdegel could you let me know what operative system are you using?

**Comment by Ryder25** [comment]
Try downloading, and installing .Net Framework 4.7.1

https://dotnet.microsoft.com/download/dotnet-framework

**Comment by Filipe-Misael** [comment]
> 
> 
> Try downloading, and installing .Net Framework 4.7.1
> 
> https://dotnet.microsoft.com/download/dotnet-framework

Thank you for the reply!
Unfortunately it's not working for me, the installer gives me this message:

".NET Framework 4.7.1 or a later update is already installed on this computer."

I was considering trying to uninstall net framework just to see if this version would install, but I'm not quite sure how to do that (can't find .NetFramework on the usual "program uninstall window") neither am I sure if I should uninstall it... =P

**Comment by Ryder25** [comment]
Can you check the Windows logs in the Event Viewer and see if there is any log entry for this app?

If so, copy and paste the error text that it shows.

**Comment by Filipe-Misael** [comment]
Alright, so I checked "Event Viewer (Local)" > Windows Logs > Application

When the software crashes, I get **3 "Error"** messages and **1 "Information"** message

It's quite a bit of text, but I'm gonna separate it by each error message (from top to bottom)

**"Information" message**
```
Fault bucket 2106590656907254626, type 4
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: Ego PSSG Editor.exe
P2: 11.1.2017.802
P3: 5982bbed
P4: d3d9.dll
P5: 10.0.18362.329
P6: 1ac9a115
P7: c000001d
P8: 00000000000acd21
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9860.tmp.mdmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER993C.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER994D.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER994D.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER996D.tmp.txt

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_Ego PSSG Editor._3567d86f9e195f9fe404997f03b3dd0694bcbca_addc1c7c_65196f43-45dd-4fc5-b2a7-faa5d480e385

Analysis symbol: 
Rechecking for solution: 0
Report Id: 070b828e-c48b-4f42-a082-4ff3f048e43b
Report Status: 268435456
Hashed bucket: 4cb3c0c87f41cf081d3c1d0a901d1b62
Cab Guid: 0
```


**Error (1)**
```
Windows cannot access the file  for one of the following reasons: there is a problem with the network connection, the disk that the file is stored on, or the storage drivers installed on this computer; or the disk is missing. Windows closed the program Ego PSSG Editor because of this error.

Program: Ego PSSG Editor
File: 

The error value is listed in the Additional Data section.
User Action
1. Open the file again. This situation might be a temporary problem that corrects itself when the program runs again.
2. If the file still cannot be accessed and
	- It is on the network, your network administrator should verify that there is not a problem with the network and that the server can be contacted.
	- It is on a removable disk, for example, a floppy disk or CD-ROM, verify that the disk is fully inserted into the computer.
3. Check and repair the file system by running CHKDSK. To run CHKDSK, click Start, click Run, type CMD, and then click OK. At the command prompt, type CHKDSK /F, and then press ENTER.
4. If the problem persists, restore the file from a backup copy.
5. Determine whether other files on the same disk can be opened. If not, the disk might be damaged. If it is a hard disk, contact your administrator or computer hardware vendor for further assistance.

Additional Data
Error value: 00000000
Disk type: 0
```


**Error (2)**
```
Faulting application name: Ego PSSG Editor.exe, version: 11.1.2017.802, time stamp: 0x5982bbed
Faulting module name: d3d9.dll, version: 10.0.18362.329, time stamp: 0x1ac9a115
Exception code: 0xc000001d
Fault offset: 0x00000000000acd21
Faulting process ID: 0x22b4
Faulting application start time: 0x01d57bc6cd29950a
Faulting application path: W:\Grid Autosport Modding\TOOLS\Ego PSSG Editor 11.1.2017.0802\Ego PSSG Editor\Ego PSSG Editor.exe
Faulting module path: C:\Windows\SYSTEM32\d3d9.dll
Report ID: 070b828e-c48b-4f42-a082-4ff3f048e43b
Faulting package full name: 
Faulting package-relative application ID: 
```


**Error (3)**
```
Application: Ego PSSG Editor.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: exception code c000001d, exception address 00007FFA0C94CD21
```

Hopefully that helps ^_^"
Thank you!

**Comment by Ryder25** [comment]
Hmm, maybe it's a DirectX or graphics thing.  Maybe try updating drivers?

I've also search around for the problem, and some have said turning off certain graphics overlays like from MSI Afterburner helped.

I was going to suggest putting the app in your Documents folder, but it seems you've already tried running the app from other locations.

I'm not sure what else it could be.

**Comment by Filipe-Misael** [comment]
> 
> I've also searched around for the problem, and some have said turning off certain graphics overlays like from MSI Afterburner helped.
> 
**Thank you so much for telling me about this one!** I just tried closing MSI Afterburner and **now it works!** XD

I'm extra excited about this because I've been having problems with other applications (some codemasters related) like the first Grid game, I couldn't even launch the game! I also just tried it now and it finally works!
I can't believe Afterburner has been giving me so many problems XD

Thank you so much!

(Good thing I only use Afterburner to keep an eye on GPU temps... Should be easy to find a replacement...)
